flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackend.java

```diff
@@ -84,13 +84,44 @@
 public interface StateBackend extends java.io.Serializable {
 
 	// ------------------------------------------------------------------------
-	//  Persistent Bytes Storage
+	//  Checkpoint storage - the durable persistence of checkpoint data
+	// ------------------------------------------------------------------------
+
+	/**
+	 * Resolves the given pointer to a checkpoint/savepoint into a state handle from which the
+	 * checkpoint metadata can be read. If the state backend cannot understand the format of
+	 * the pointer (for example because it was created by a different state backend) this method
+	 * should throw an {@code IOException}.
+	 *
+	 * @param pointer The pointer to resolve.
+	 * @return The state handler from which one can read the checkpoint metadata.
+	 *
+	 * @throws IOException Thrown, if the state backend does not understand the pointer, or if
+	 *                     the pointer could not be resolved due to an I/O error.
+	 */
+	StreamStateHandle resolveCheckpoint(String pointer) throws IOException;
```

Comment:
Similar to the `String` vs `StoragePointer`, this could also be a class that, essentially , is a `StreamStateHandle` but gives a name to the concept, e.g. `CheckpointMetaDataHandle`. But it is not as bad as the other case, because this is never passed down very deep (yet).

Reply:
See above, I don't think we get around String.